contracts/evmscript/EVMScriptRunner.sol

```diff
@@ -0,0 +1,60 @@
+pragma solidity ^0.4.18;
+
+import "../apps/AppStorage.sol";
+import "./IEVMScriptExecutor.sol";
+import "./IEVMScriptRegistry.sol";
+
+import "./ScriptHelpers.sol";
+
+
+contract EVMScriptRunner is AppStorage, EVMScriptRegistryConstants {
+    using ScriptHelpers for bytes;
+
+    function runScript(bytes script, bytes input, address[] blacklist) protectState internal returns (bytes output) {
+        // TODO: Too much data flying around, maybe extracting spec id here is cheaper
+        address executorAddr = getExecutor(script);
+        require(executorAddr != address(0));
+
+        bytes memory calldataArgs = script.encode(input, blacklist);
+        bytes4 sig = IEVMScriptExecutor(0).execScript.selector;
+
+        require(executorAddr.delegatecall(sig, calldataArgs));
+
+        return returnedDataDecoded();
+    }
+
+    function getExecutor(bytes script) public view returns (IEVMScriptExecutor) {
+        return IEVMScriptExecutor(getExecutorRegistry().getScriptExecutor(script));
+    }
+
+    // TODO: Internal
+    function getExecutorRegistry() internal view returns (IEVMScriptRegistry) {
+        address registryAddr = kernel.getApp(EVMSCRIPT_REGISTRY_APP);
+        return IEVMScriptRegistry(registryAddr);
+    }
+
+    /**
+    * @dev copies and returns last's call data. Needs to ABI decode first
+    */
+    function returnedDataDecoded() internal view returns (bytes ret) {
```

Comment:
We have a few implementations of returning data from a `delegatecall` (see `DelegateProxy`, this, and `DelegateScript`), but each looks somewhat different. Are there different cases for handling each or can we put this functionality into a base contract?